core/src/main/java/de/javakaffee/web/msm/StorageClientFactory.java

```diff
@@ -78,9 +87,30 @@ static MemcachedClient createCouchbaseClient(final MemcachedNodesManager memcach
             }
         }
 
+        protected static MemcachedNodeURLBasedConnectionFactory createMemcachedNodeURLBasedConnectionFactory() {
+            try {
+                Class.forName("net.spy.memcached.ClientMode");
```

Comment:
This could be extracted into a method like `isElastiCacheClientLib` (which also hides exception handling), so that here would be a simple if/else.